core/lf_utils.cmake

```diff
@@ -0,0 +1,11 @@
+function(lf_enable_compiler_warnings target)
+    if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+        target_compile_options(${target} PRIVATE -Wall -Wextra -Wpedantic -Werror)
+    elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
+        target_compile_options(${target} PRIVATE /W4)
+    elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+        target_compile_options(${target} PRIVATE -Wall)
+    else()
+        target_compile_options(${target} PRIVATE -Wall)
+    endif()
+endfunction()
```

Comment:
This seems to omit some of commands it replaces, which are these:
```
if(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
    set(CMAKE_SYSTEM_VERSION 10.0)
    message("Using Windows SDK version ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION}")
elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Nrf52")
    list(APPEND REACTORC_COMPILE_DEFS PLATFORM_NRF52)
elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Zephyr")
    list(APPEND REACTORC_COMPILE_DEFS PLATFORM_ZEPHYR)
    set(PLATFORM_ZEPHYR true)
elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Rp2040")
    list(APPEND REACTORC_COMPILE_DEFS PLATFORM_RP2040)
endif()
```
Just checking to make sure this was deliberate.